### PR TITLE
Store domains in a database

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following:**

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/AdminLTE/blob/master/CONTRIBUTING.md), as well as this entire template.
- [X] I have made only one major change in my proposed changes.
- [X] I have commented my proposed changes within the code.
- [X] I have tested my proposed changes.
- [X] I am willing to help maintain this change if there are issues with it later.
- [X] I give this submission freely and claim no ownership.
- [X] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [X] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
- [X] I have Signed Off all commits. ([`git commit --signoff`](https://git-scm.com/docs/git-commit#git-commit---signoff))

---

**What does this PR aim to accomplish?:**

Frontend implementation of changes in https://github.com/pi-hole/pi-hole/pull/2611

**How does this PR accomplish the above?:**

Read from gravity's database instead of from the list files. Adding/deleting is still done through the corresponding `pihole ...` callbacks. This will be addressed by the follow-up API which will be part of Pi-hole v6.0

**What documentation changes (if any) are needed to support this PR?:**

This is a breaking change and will bring Pi-hole to version 5.0

> * `{Please delete this quoted section when opening your pull request}`
> * You must follow the template instructions. Failure to do so will result in your issue being closed.
> * Please respect that Pi-hole is developed by volunteers, who can only reply in their spare time.
> * Detail helps us understand an issue quicker, but please ensure it's relevant.

### CODE REVIEW

**Comment by DL6ER** [comment]
This PR only partially implements future capabilities of our new domain management. This is intentional as the specific implementation will be left open for the upcoming rewritten [web](https://github.com/pi-hole/web) interface.

Features available in this version:
- Disabled entries are highlighted in gray. However, the status cannot be changed from the dashboard.
  Screenshot at 2019-04-25 15-31-50
  Displaying the current status is still meaningful as `pihole-FTL` already ignores disabled entries even if we do not provide any way of disabling domains yet.


- Comments are shown. However, comments cannot be changed from the dashboard.
  Screenshot at 2019-04-25 15-46-58

- We store the time/date when a new entry is added in the database and show this information when hovering the domain.
  Screenshot at 2019-04-25 15-52-58


---
Note: The status and comments have been set for demonstration purposes using
```bash
sqlite3 /etc/pihole/gravity.db "UPDATE blacklist SET enabled = 0 WHERE domain='black2.com';"
sqlite3 /etc/pihole/gravity.db "UPDATE blacklist SET comment = 'disabled for demonstration purposes' WHERE domain='black2.com';"
sqlite3 /etc/pihole/gravity.db "UPDATE blacklist SET comment = 'some other comment' WHERE domain='black3.net';"
```

File: scripts/pi-hole/php/add.php [hunk 1]
```diff
@@ -22,36 +22,36 @@
...
 switch($type) {
     case "white":
         if(!isset($_POST["auditlog"]))
-            echo shell_exec("sudo pihole -w ${_POST['domain']}");
+            echo shell_exec("sudo pihole -w --web ".$domains);
         else
         {
-            echo shell_exec("sudo pihole -w -n ${_POST['domain']}");
-            echo shell_exec("sudo pihole -a audit ${_POST['domain']}");
+            echo shell_exec("sudo pihole -w --web -n ".$domains);
```
Comment (Mcat12) [inline scripts/pi-hole/php/add.php]: I don't see an `-n` flag in `list.sh`, perhaps it was removed at some point?

File: scripts/pi-hole/php/get.php [hunk 2]
```diff
@@ -11,57 +11,100 @@
...
+if(isset($FTLsettings["GRAVITYDB"]))
+{
+	$GRAVITYDB = $FTLsettings["GRAVITYDB"];
+}
+else
+{
+	$GRAVITYDB = "/etc/pihole/gravity.db";
 }
 
+function SQLite3_connect($dbfile, $trytoreconnect)
```
Comment (Mcat12) [inline scripts/pi-hole/php/get.php]: This function is duplicated here from `api_db.php`, can it be moved to a common file and imported?

File: scripts/pi-hole/js/list.js [hunk 3]
```diff
@@ -62,53 +36,68 @@ function refresh(fade) {
...
+                    data = response.blacklist.sort();
+                    data2 = response.regex.sort();
+                }
+                data.forEach(function (entry, index)
+                {
+                    var used = entry.enabled === "1" ? "used" : "not-used";
+                    var comment = entry.comment.length > 0 ? "&nbsp;-&nbsp;" + entry.comment : "";
+                    var time = new Date(parseInt(entry.date_added)*1000);
+                    var added = "Added: " + time.toLocaleString();
+                    console.log(time);
```
Comment (Mcat12) [inline scripts/pi-hole/js/list.js]: I assume this was for debugging and can be removed?

**Comment by Mcat12** [comment]
Error when opening the settings page:
image

**Comment by Mcat12** [comment]
The adlists section of the settings page needs to be converted to use the database.

**Comment by DL6ER** [comment]
The issue with the settings page has been fixed in ced7174

Screenshot at 2019-04-28 21-46-49

File: settings.php [hunk 4]
```diff
@@ -283,7 +291,7 @@
                                             </table>
                                         </div>
                                         <div class="form-group">
-                                            <textarea name="newuserlists" class="form-control" rows="1" placeholder="Enter one URL per line to add new blocklists"></textarea>
+                                            <input name="newuserlists" class="form-control" placeholder="Enter URL to add a new blocklists"></input>
```
Comment (Mcat12) [inline settings.php]: With the change in plurality, `blocklists` should be `blocklist`, and `a` should be inserted before `URL`

**Pull request merged**